PropertyChanged.Fody/CecilExtensions.cs

```diff
@@ -11,6 +11,12 @@ public static string GetName(this PropertyDefinition propertyDefinition)
         return $"{propertyDefinition.DeclaringType.FullName}.{propertyDefinition.Name}";
     }
 
+    public static bool IsCallToMethodFromThis(this Instruction instruction, MethodDefinition method)
+    {
+        return instruction.IsCallToMethod(method) 
+            && instruction.Previous?.Previous?.OpCode == OpCodes.Ldarg_0;
```

Comment:
That `Previous.Previous` call won't return what you want if the weaved code does something like:

```C#
base.Prop = Some().Nontrivial().Expression().Here();
```

See [example here](https://sharplab.io/#v2:C4LglgNgPgAgTARgLAChUwMwAJ5YEJYDeqWpO2AbmAE7ACuAhhFmAHbBYAK1A9gA5EA5gFNgAbgDOosQF9UctCkw44WAMJYQ+IiTLKeFYdWpgAJsJbsuvPrtLEUZJ1hEcAvAD4sAIwZSAdNz8YnbOUu5evgFBAm5YrMIA7lgASgyspjwAtgAUFEx0wgCU/gBywgAewDlFIY5kCjJAA==) in `C.set_Prop`

Reply:
Right, forgot about that...

Reply:
I guess the easiest fix here would be to check the opcode - it will be `call` for a call to the base method (implying `this`), and `callvirt` for everything else since the method is virtual.

Reply:
Wasn's sure if that call/callvirt is true for every case, and I already had that stack balance loop ready from another project, so `IsCallToMethodFromThis` is now universal and really does what it's name implies.

Reply:
Yeah, I have similar code in InlineIL as well, but I'm not sure the added complexity is worth it here.
Besides, I _suppose_ it won't work for calls like these (but I haven't checked):

```C#
base.Prop = value > 0 ? value : throw new InvalidOperationException();
```

I'd prefer a simple opcode check because of this. I'm pretty sure it should be reliable, and `IsCallToMethodFromThis` could be renamed 😉

Reply:
Right, that works better...